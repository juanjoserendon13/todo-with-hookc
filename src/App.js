import React, { useState } from "react";
import Form from "./components/Form";

import "./App.css";

const App = () => {
  const [todos, setTodos] = useState([]);
  const toggleComplete = i => {
    setTodos(
      todos.map((todo, k) =>
        k === i
          ? {
              ...todo,
              complete: !todo.complete
            }
          : todo
      )
    );
  };
  return (
    <div className="App">
      <Form
        onSubmit={text => setTodos([{ text, complete: false }, ...todos])}
      />
      <div>
        {todos.map(({ text, complete }, i) => (
          <div
            id="todo"
            className="todo"
            key={Math.random() * 10}
            onClick={() => toggleComplete(i)}
            style={{
              textDecoration: complete ? "line-through" : ""
            }}
          >
            {text}
          </div>
        ))}
      </div>
      <button onClick={() => setTodos([])}>Reset</button>
    </div>
  );
};

export default App;
