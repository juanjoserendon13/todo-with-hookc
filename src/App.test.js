import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import App from "./App";
import Form from "./components/Form";

Enzyme.configure({ adapter: new Adapter() });

describe("App", () => {
  it("render app component in isolated mode", () => {
    // The shallow function allow to render the
    // component without including its children
    const appComponent = shallow(<App debug />);
    expect(appComponent).toMatchSnapshot();
  });

  it("add todo when submit and click the todo", () => {
    const appComponent = mount(<App />);
    const form = appComponent.find("form");
    const input = appComponent.find("input");
    let prevented = false;
    input.simulate("change", { target: { value: "new todo" } });
    form.simulate("submit", { preventDefault: () => (prevented = true) });

    const todo = appComponent.find("div.todo");

    // expect(input.props().value).toEqual("new todo");
    expect(prevented).toBe(true);
    expect(todo.length).toBe(1);
    expect(todo.text()).toBe("new todo");

    console.log(todo.debug());
    todo.simulate("click");
    appComponent.update();
    expect(todo.props().style).toHaveProperty("textDecoration", "line-through");
  });
});
