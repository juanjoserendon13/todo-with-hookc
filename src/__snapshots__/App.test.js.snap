// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render app component in isolated mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Form
          onSubmit={[Function]}
        />,
        <div />,
        <button
          onClick={[Function]}
        >
          Reset
        </button>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Reset",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Form
            onSubmit={[Function]}
          />,
          <div />,
          <button
            onClick={[Function]}
          >
            Reset
          </button>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Reset",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
